app/src/main/java/com/example/rickandmorty/fragment/ListFragment.kt

```diff
@@ -0,0 +1,153 @@
+                .setOnMenuItemClickListener {
+                    showDialog()
+                    true
+                }
+
+            layoutSwiperefresh.setOnRefreshListener {
+                adapter.submitList(emptyList())
+                currentPage = 0
+                loadCharacters {
+                    layoutSwiperefresh.isRefreshing = false
+                }
+            }
+
+            val linearLayoutManager = LinearLayoutManager(
+                view.context, LinearLayoutManager.VERTICAL, false
+            )
+
+            recyclerView.adapter = adapter
+            recyclerView.layoutManager = linearLayoutManager
+            recyclerView.addHorizontalSpaceDecoration(ITEM_SPACE)
+            recyclerView.addPaginationScrollListener(linearLayoutManager, ITEMS_TO_LOAD) {
+                loadCharacters()
+            }
+
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        currentRequest?.cancel()
+        _binding = null
+    }
+
+    private fun loadCharacters(onLoadingFinished: () -> Unit = {}) {
+        if (isLoading) return
+
+        val loadingFinishedCallback = {
+            isLoading = false
+            onLoadingFinished()
+        }
+
+        val since = currentPage
+
+        currentRequest = RickAndMortyService.rickAndMortyApi.getCharacters(since)
+        currentRequest?.enqueue(object : Callback<CharacterList> {
+            override fun onResponse(
+                call: Call<CharacterList>,
+                response: Response<CharacterList>
+            ) {
+                if (response.isSuccessful) {
+                    val newList = adapter.currentList
+                        .dropLastWhile { it == PagingData.Loading }
+                        .plus(response.body()?.results?.map { PagingData.Content(it) }.orEmpty())
+                        .plus(PagingData.Loading)
+                    adapter.submitList(newList)
+                    currentPage++
+
+                } else {
+                    handleErrors(response.errorBody()?.string() ?: ERROR_MESSAGE)
+                }
+                loadingFinishedCallback()
+                currentRequest = null
+            }
+
+            override fun onFailure(call: Call<CharacterList>, t: Throwable) {
+                handleErrors(t.message ?: ERROR_MESSAGE)
+                loadingFinishedCallback()
+                currentRequest = null
+            }
+        })
+    }
+    private fun handleErrors(errorMessage: String) {
+        Snackbar.make(binding.root, errorMessage, Snackbar.LENGTH_SHORT)
+            .setAction(android.R.string.ok) {}
+            .show()
+    }
+
+    private  fun showDialog(){
+        AlertDialog.Builder(requireContext())
+            .setTitle("Information")
+            .setMessage("Click on a character in the list to see more information about him")
+            .setPositiveButton(android.R.string.ok){dialog, buttonId ->
+
+            }
```

Comment:
можно передать null в качестве колбека 
```suggestion
            .setPositiveButton(android.R.string.ok, null)
```